addons/field_rations/XEH_postInit.sqf

```diff
@@ -74,26 +74,31 @@ if !(hasInterface) exitWith {};
     } forEach _subActions;
 
     // Add inventory context menu option to consume items
-    ["ACE_ItemCore", ["CONTAINER"], LSTRING(EatDrink), [], QPATHTOF(ui\icon_survival.paa),
-        [
-            {true},
-            {
-                params ["", "", "_item"];
+    private _eatOrDrinkCondition = [
+        {true},
+        {
+            params ["", "", "_item"];
 
-                XGVAR(enabled) && {
-                    private _config = configFile >> "CfgWeapons" >> _item;
-                    getNumber (_config >> QXGVAR(thirstQuenched)) > 0
-                    || {getNumber (_config >> QXGVAR(hungerSatiated)) > 0}
-                }
+            XGVAR(enabled) && {
+                private _config = _item call CBA_fnc_getItemConfig;
+                getNumber (_config >> QXGVAR(thirstQuenched)) > 0
+                || {getNumber (_config >> QXGVAR(hungerSatiated)) > 0}
             }
-        ],
-        {
-            params ["_unit", "", "_item"];
-            private _itemConfig = configFile >> "CfgWeapons" >> _item;
-            [objNull, _unit, [_item, _itemConfig, false]] call FUNC(consumeItem);
-            false
         }
-    ] call CBA_fnc_addItemContextMenuOption;
+    ];
+    private _eatOrDrinkStatement = {
+        params ["_unit", "", "_item"];
+        private _itemConfig = _item call CBA_fnc_getItemConfig;
+        private _isMagazine = _item isKindOf ["CA_Magazine", configFile >> "CfgMagazines"];
```

Comment:
I couldn't find if there was some way to check if a given class was a part of a config path, so I went with just an `isKindOf` check. You could do it by turning it into a string but that feels hacky

Reply:
Just use `isClass (configFile >> "CfgMagazines" >> _item)`.

Reply:
True, that makes a lot more sense. Kinda forgot about isClass